services/supervisor/src/mode.rs

```diff
@@ -27,11 +27,15 @@ use exonum::helpers::byzantine_quorum;
 use exonum_crypto::Hash;
 use exonum_merkledb::access::Access;
 use exonum_proto::ProtobufConvert;
+use failure::{self, format_err};
+
+use std::str::FromStr;
 
 use super::{multisig::MultisigIndex, proto, DeployRequest};
 
 /// Supervisor operating mode.
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
```

Comment:
Nit / comment: Since `Copy` (non)implementation significantly influences the public interface of a type, we should be careful about implementing `Copy` for public types that may be extended in the future (of which `Mode` is a good example).